File: 674-longest-continuous-increasing-subsequence/674-longest-continuous-increasing-subsequence.rb
# @param {Integer[]} nums
# @return {Integer}
def find_length_of_lcis(nums)
    length = 1
    max_length = 1
    nums.each_with_index do |num,i|
        next if i == 0
        if num > nums[i-1]
            length += 1
            max_length = length if length > max_length
        else
            length = 1
        end
    end
    max_length
        
end

File: 674-longest-continuous-increasing-subsequence/674-longest-continuous-increasing-subsequence.py
class Solution:
    def findLengthOfLCIS(self, nums: List[int]) -> int:
        subseq = [nums[0]]
        res = 1
        
        for num in nums[1:]:
            if num>subseq[-1]:
                subseq.append(num)
                res=max(res,len(subseq))
            else:
                subseq=[num]
                
        return res